### PR TITLE
Better/more automated deployment process

### PR DESCRIPTION
**Pull request by iainnash**
1. Separate `chainConfigs` and `addresses` folders
2. Create a new base class to read and deploy updates.
3. Handle writing updates between `run-latest.json` and `addresses/` folder.

### CODE REVIEW

File: chainConfigs/1.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (oveddan) [inline chainConfigs/1.json:1]: looks good! the deploy script still needs to be updated to read from this, right?

Comment (iainnash) [reply chainConfigs/1.json:1]: yep!

File: script/DeployFeeRegistry.s.sol [hunk 2]
```diff
@@ -1,68 +1,32 @@
...
-        uint256 chainID = vm.envUint("CHAIN_ID");
-        console.log("CHAIN_ID", chainID);
-
         console2.log("Starting ---");
 
+        
+
         vm.startBroadcast();
-        address subscriptionOwner = vm.envAddress("SUBSCRIPTION_OWNER");
+        address subscriptionOwner = getChainConfig().subscriptionMarketFilterOwner;
```
Comment (oveddan) [inline script/DeployFeeRegistry.s.sol:25]: can we move this fetch outside of the broadcast?

File: script/UpgradeERC721DropFactory.s.sol [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            0,
+            payable(address(0)),
+            address(0),
+            saleConfig,
+            "",
+            "ipfs://bafkreigu544g6wjvqcysurpzy5pcskbt45a5f33m6wgythpgb3rfqi3lzi",
+            "ipfs://bafkreigu544g6wjvqcysurpzy5pcskbt45a5f33m6wgythpgb3rfqi3lzi"
+        ));
+
+        console2.log("Deploying new contract for verification purposes", newContract);
```
Comment (oveddan) [inline script/UpgradeERC721DropFactory.s.sol:79]: this console.log belongs above the previous line, correct?

File: script/ZoraDropsDeployBase.sol [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        dropDeployment.editionMetadata = json.readAddress(".EDITION_METADATA_RENDERER");
+        dropDeployment.dropImplementation = json.readAddress(".ERC721DROP_IMPL");
+        dropDeployment.factoryUpgradeGate = json.readAddress(".FACTORY_UPGRADE_GATE");
+        dropDeployment.factory = json.readAddress(".ZORA_NFT_CREATOR_PROXY");
+        dropDeployment.factoryImpl = json.readAddress(".ZORA_NFT_CREATOR_V1_IMPL");
+    }
+
+    function writeDeployment(DropDeployment memory deployment) internal {
+        string memory deploymentJson = "{}";
+        deploymentJson = vm.serializeAddress(deploymentJson, "DROP_METADATA_RENDERER", deployment.dropMetadata);
```
Comment (oveddan) [inline script/ZoraDropsDeployBase.sol:57]: maybe we can have these keys come from `const`

File: script/Deploy.s.sol [hunk 5]
```diff
@@ -3,94 +3,78 @@ pragma solidity ^0.8.13;
...
-            0,
-            payable(address(0)),
-            address(0),
-            saleConfig,
-            "",
-            "ipfs://bafkreigu544g6wjvqcysurpzy5pcskbt45a5f33m6wgythpgb3rfqi3lzi",
-            "ipfs://bafkreigu544g6wjvqcysurpzy5pcskbt45a5f33m6wgythpgb3rfqi3lzi"
-        ));
+        address newContract = address(
+            factory.createEdition(
```
[the remark this answers is not in the document]
Comment (oveddan) [reply script/Deploy.s.sol:54]: if its for verification, maybe we can have some fork tests

[the remark this answers is not in the document]
Comment (iainnash) [reply script/Deploy.s.sol:54]: it's for verification – fork tests won't work we need to deploy to mainnet for etherscan to pick up the address

File: script/ZoraDropsDeployBase.sol [hunk 6]
```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "forge-std/Script.sol";
+import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
+import {ZoraNFTCreatorV1} from "../src/ZoraNFTCreatorV1.sol";
+import {IERC721Drop} from "../src/interfaces/IERC721Drop.sol";
+
+struct ChainConfig {
```
Comment (oveddan) [inline script/ZoraDropsDeployBase.sol:9]: how about adding natspec comments to this?

File: chainConfigs/11155111.json [hunk 7]
```diff
@@ -0,0 +1,9 @@
+{
+  "FACTORY_OWNER": "0xE51051a204afb2fC071A2406914cdEA5638e5018",
+  "FACTORY_UPGRADE_GATE_OWNER": "0xE51051a204afb2fC071A2406914cdEA5638e5018",
+  "MINT_FEE_AMOUNT": 111000000000000,
+  "MINT_FEE_RECIPIENT": "0xE51051a204afb2fC071A2406914cdEA5638e5018",
+  "SUBSCRIPTION_MARKET_FILTER_ADDRESS": "0x9B414A3F7872bdd2E6513689214BD2Debbe48340",
+  "SUBSCRIPTION_MARKET_FILTER_OWNER": "0xE51051a204afb2fC071A2406914cdEA5638e5018",
+  "ZORA_ERC721_TRANSFER_HELPER": "0xd1adAF05575295710dE1145c3c9427c364A70a7f"
```
Comment (oveddan) [inline chainConfigs/11155111.json:8]: this was a 0 address before

**Review by oveddan — APPROVED** [review]

**Pull request merged**